sdk/core/System.ClientModel/gen/TypeSymbolTypeRefCache.cs

```diff
@@ -55,6 +62,32 @@ private TypeRef FromTypeSymbol(ITypeSymbol symbol, TypeSymbolKindCache symbolToK
             }
         }
 
+        private IAssemblySymbol GetArrayAssembly(IArrayTypeSymbol arrayTypeSymbol)
+        {
+            while (arrayTypeSymbol.ElementType is IArrayTypeSymbol innerArray)
+            {
+                arrayTypeSymbol = innerArray;
+            }
+
+            return arrayTypeSymbol.ElementType.ContainingAssembly;
+        }
+
+        private TypeRef? GetContextType(IAssemblySymbol assembly, TypeSymbolKindCache symbolToKindCache)
+        {
+            foreach (var attribute in assembly.GetAttributes())
+            {
+                if (attribute.AttributeClass?.ToDisplayString() == "System.ClientModel.Primitives.ModelReaderWriterContextTypeAttribute")
+                {
+                    if (attribute.ConstructorArguments.Length > 0 &&
+                        attribute.ConstructorArguments[0].Value is ITypeSymbol typeSymbol)
```

Comment:
You could simplify this, since you know the constructor only has the one argument:
```suggestion
                    if (attribute.ConstructorArguments is [{ Value: ITypeSymbol typeSymbol }])
```